### ISSUE TITLE
GE RDF HDF5 remaining issues

### ISSUE THREAD

**Issue by KrisThielemans** [UCL/STIR#555]
After #390 there's a few more things to do.
- [ ] various hard-wired numbers for the Signa. After fixing those, we should rename Signa files to something more generic
-  [ ] `SinglesRatesFromGEHDF5` 
    - expose `GEHDF5Wrapper.get_exam_info_sptr()`
    - `construct_randoms_from_GEsingles.cxx` needs to reorder data. This should be moved to this class
- [ ] `GESignaListmodeInputFileFormat` only checks if it's an HDF5 file
- [ ] `ProjData` should call `ProjDataGEHDF5`
- [ ] add tests (needs data)

**Comment by evgueni-ovtchinnikov** [comment]
Another one:
```
hsize_t dims_out[dataset_list_Ndims];
```
is wrong (`dataset_list_Ndims` is not a constant), should be
```
hsize_t* dims_out = new hsize_t[dataset_list_Ndims];
```
(and `delete[] dims_out; ` before return).

**Comment by AnderBiguri** [comment]
>     * [ ]  `SinglesRatesFromGEHDF5`
>       
>       * expose `GEHDF5Wrapper.get_exam_info_sptr()`
You mean  creating `SinglesRatesFromGEHDF5::get_exam_info_sptr()`? `GEHDF5Wrapper.get_exam_info_sptr()` is exposed and read in that class.

>     * [ ]  in `ProjDataGEHDF5` avoid reading all views up-front. could read them when necessary

You mean read them in `get_viewgram()/get_sinogram()` instead of in `initialise_viewgram_buffer`?

**Comment by KrisThielemans** [comment]
> You mean creating SinglesRatesFromGEHDF5::get_exam_info_sptr()? GEHDF5Wrapper.get_exam_info_sptr() is exposed and read in that class.

not sure what I was thinking at the time. Possibly because we were (erronuously) using `TimeFrameDefinitions` then, which sit in `exam_info`. Let's ignore this. (please edit the first box and remove from the list)

> you mean read them in get_viewgram()/get_sinogram() instead of in initialise_viewgram_buffer?

yes. it could be done by making `tof_data `mutable`, and fill it when necessary. However, this will have little impact, and likely will conflict with updates on the TOF branch, so I'd postpone this.

**Comment by KrisThielemans** [comment]
on
> construct_randoms_from_GEsingles.cxx needs to reorder data. This should be moved to this class
this is about
https://github.com/KrisThielemans/STIR-1/blob/7c5ca0aa99b60f01db35d3d3a0904694d3ffb0ed/src/utilities/construct_randoms_from_GEsingles.cxx#L244

This reversal of `efficiencies` should be moved to `SinglesRatesFromGEHDF5`. Once that's done, there might be very little GE specific in `construct_randoms_from_GEsingles` (I'll have to check).

**Comment by AnderBiguri** [comment]
I have now added some examples. 

Missing from that work: 

Any change related to the attenuation coefficients. I don't know where they are in the data I have nor how to read them. 

Testing it with real data and real `.fdef`

**Comment by AnderBiguri** [comment]
Added tests (and tested them!)
